bundle-workflow/python/manifests/build_manifest.py

```diff
@@ -0,0 +1,81 @@
+# Copyright OpenSearch Contributors.
+# SPDX-License-Identifier: Apache-2.0
+
+import yaml
+
+'''
+A BuildManifest is an immutable view of the outputs from a build step
+The manifest contains information about the product that was built (in the `build` section),
+and the components that made up the build in the `components` section.
+
+The format for schema version 1.0 is:
+schema-version: 1.0
+build:
+  name: string
+  version: string
```

Comment:
We should include platform/architecture here as well

Reply:
Yes, we should.

Reply:
Should we add platform as well? linux/mac/windows?

Reply:
Graah, yeah, platform is not the same as architecture. My bad :)

I have a few changes stacked on top of this...can I add platform as a new PR?

Reply:
Sure if you remember it! :)